### PR TITLE
Stop requiring use PackageDirName argument in Update-PkgVersion.ps1

### PR DESCRIPTION
**Pull request by chidozieononiwu**
- We used `PackageDirName` for packages where were the packagename is not the same as the package directory name.
-  Since we can get all the package properties we no longer need to do that.

### CODE REVIEW

**Review by weshaggard — APPROVED** [review]
Looks reasonable please verify that it works on one of the projects we used to require it.

File: eng/pipelines/templates/stages/archetype-net-release.yml [hunk 1]
```diff
@@ -215,7 +215,7 @@ stages:
                     steps:
                       - checkout: self
                       - pwsh: |
-                          eng/scripts/Update-PkgVersion.ps1 -ServiceDirectory '${{parameters.ServiceDirectory}}' -PackageName '${{artifact.name}}' -PackageDirName '${{artifact.directoryName}}'
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-net-release.yml:218]: Can we remove artifact.directoyrName from the repo now as well or is it used for other things?

Comment (chidozieononiwu) [reply eng/pipelines/templates/stages/archetype-net-release.yml:218]: I think we still use it for some other things. e.g. in the github.io doc generation process.

Comment (weshaggard) [reply eng/pipelines/templates/stages/archetype-net-release.yml:218]: OK we can keep it for now but eventually I'd like to get the other process using the same properties.

**Pull request merged**